streams/shared/src/main/scala/zio/stream/ZStream.scala

```diff
@@ -1792,23 +1795,22 @@ object ZStream extends ZStreamPlatformSpecific {
    */
   final def fromChunk[@specialized A](c: Chunk[A]): Stream[Nothing, A] =
     new Stream[Nothing, A] {
-      def fold[R1, E1, A1 >: A, S]: Fold[R1, E1, A1, S] =
-        ZManaged.succeed { (s, cont, f) =>
-          ZManaged.fromEffect(c.foldMLazy(s)(cont)(f))
+      def fold[R, E, A1 >: A, S]: Fold[R, E, A1, S] = foldDefault
+
+      override def process: Managed[Nothing, InputStream[Nothing, A]] =
+        for {
+          ref <- Ref.make(c).toManaged_
+        } yield ref.get.flatMap { chunk =>
```

Comment:
Slightly more efficient:
- read the chunk length
- keep the last read index in the ref
- bump it on every pull

`chunk.drop(1)` will allocate a new `Chunk.Slice` on every pull

Reply:
Basically keep the index in a Ref?

Reply:
Yup!

Reply:
Done.